improver/psychrometric_calculations/hail_size.py

```diff
@@ -0,0 +1,392 @@
+        assert_spatial_coords_match([ccl_temperature, ccl_pressure, temp_slice])
+        assert_spatial_coords_match(
+            [temperature_on_pressure, relative_humidity_on_pressure]
+        )
+
+        ccl_temperature.convert_units("K")
+        ccl_pressure.convert_units("Pa")
+        temperature_on_pressure.convert_units("K")
+        relative_humidity_on_pressure.convert_units("kg/kg")
+
+    def variable_at_pressure(
+        self, variable_on_pressure: Cube, pressure: np.ndarray
+    ) -> np.ndarray:
+        """Extracts the values from variable_on_pressure cube at pressure
+        levels described by the pressure array"""
+
+        variable = []
+        pressure_grid = self.pressure_grid(variable_on_pressure)
+
+        # deals with each realization independently and puts the arrays back together
+        for rea in range(len(pressure_grid)):
+            pressure_diff = abs(pressure_grid[rea] - pressure[rea])
+
+            indicies = np.nanargmin(pressure_diff, axis=0)
+            lat, long = indicies.shape
+            lat, long = np.ogrid[:lat, :long]
+
+            variable.append(variable_on_pressure.data[rea, indicies, lat, long])
+
+        return np.array(variable)
+
+    def pressure_grid(self, variable_on_pressure: Cube) -> np.ndarray:
+        """Creates a pressure_grid of the same shape as variable_on_pressure
+        with all possible pressure levels."""
+
+        required_shape = variable_on_pressure.shape
+        pressure_points = variable_on_pressure.coord("pressure").points
+        (pressure_axis,) = variable_on_pressure.coord_dims("pressure")
+        pressure_shape = np.ones_like(required_shape)
+        pressure_shape[pressure_axis] = required_shape[pressure_axis]
+        pressure_array = np.broadcast_to(
+            pressure_points.reshape(pressure_shape), required_shape
+        )
+        return pressure_array
+
+    def pressure_at_268(self, temperature_on_pressure: Cube) -> (CubeList):
+        """Extracts the lowest pressure level where the environment
+        temperature is below -5 degrees (268.15K). It also produces
+        the environemnt temperature at that pressure value"""
+
+        pressure_template = next(temperature_on_pressure.slices_over(["pressure"]))
+        pressure_template.rename("pressure of atmosphere at 268.15K")
+        pressure_template.units = temperature_on_pressure.coord("pressure").units
+        pressure_template.remove_coord("pressure")
+
+        temperature_template = next(temperature_on_pressure.slices_over(["pressure"]))
+        temperature_template.rename("tempeature_of_atmosphere_at_268.15K")
+        temperature_template.remove_coord("pressure")
+
+        shape = temperature_on_pressure.data.shape
+        r, l, m = (shape[0], shape[2], shape[3])  # ignores pressure axis
+
+        data = np.ma.masked_greater(temperature_on_pressure.data, 268.15)
+        data = np.ma.masked_invalid(data)
+
+        indicies = np.ma.notmasked_edges(data, axis=1)[0][1]
```

Comment:
You've inherited my common typo. This should be `indices` everywhere.
```suggestion
        indices= np.ma.notmasked_edges(data, axis=1)[0][1]
```

Reply:
There are still quite a lot of `indicies` typos in this code. There are now also `indexes`, just to use both plural options. I'm all for diversity, so I guess this is fine, but we should probably fix the typos.

Reply:
I think I've managed to find all the indicies typos.